datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -710,6 +714,16 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
       retryHandleAssignmentChange(isDatastreamUpdate);
     }
 
+    if (_config.getEnableAssignmentTokens()) {
+      try {
+        // Queue assignment token claim task
+        _executor.schedule(() -> maybeClaimAssignmentTokensForStoppingStreams(newAssignment, oldAssignment),
```

Comment:
This `_executor` object is used for CoordinatorEventQueue. Calling this API using the same executor and any delay in this can result in a delay in a heartbeat and other operations. Was this intentional?

Reply:
This was intentional. I didn't want to put the code on the thread that executes `handleAssignmentChange` and I wanted to make the checks for stopped tasks and token claims asynchronous. `maybeClaimAssignmentTokensForStoppingStreams` has a blocking wait for tasks to stop. What would be a better way to execute this on a separate thread? Do you want me to introduce a new executor service?

Reply:
A new executor service.